### DISCUSSION TITLE
Replace hardcoded SPI pin numbers by variable/parameter?

### DISCUSSION THREAD

**Discussion by martybillingsley** [Bodmer/TFT_eSPI#2612] [New processor/display support or capability request]
At runtime, I'd like to be able to choose which SPI pins the TFT display is plugged into. Particularly TFT_CS and TFT_DC, but flexibility on some of the other pins would also be nice. Currently these are defined in User_Setup.h and seem to be immutable after compilation.

Is there a way to use a variable or parameter instead of #define for each pin so that I can select the pins at runtime, when a new instance of TFT_eSPI is created?

**Comment by Bodmer** [comment]
The library does not support that capability and adding it is not planned.

**Reply by martybillingsley** [reply]
I might give it a go. Any hints or tips? Are the pin numbers used only when TFT_eSPI is instantiated? Or are they also used after that? (in which case it might be necessary to add the pin numbers as member variables to the TFT_eSPI class)  I see that they are used in the initDMA function, but don't see who actually calls that function. I'll keep poking through the code, but pointers would be appreciated.
